src/app/categories/categories.component.ts

```diff
@@ -0,0 +1,34 @@
+import { Component, OnInit } from '@angular/core';
+import { CategoriesListItem } from './model';
+import { CATEGORIES } from './mock-data';
+import { Ng4LoadingSpinnerService } from 'ng4-loading-spinner';
+
+@Component({
+  selector: 'app-categories',
+  templateUrl: './categories.component.html',
+  styleUrls: ['./categories.component.scss']
+})
+export class CategoriesComponent implements OnInit {
+  categoriesItems: CategoriesListItem[] = [];
+
+  constructor(private spinnerService: Ng4LoadingSpinnerService) { }
+
+  ngOnInit() {
+    //show spinner loading 
+    this.spinnerService.show();
```

Comment:
@ikramabumazen  may be moving the show/hide spinner , and also the timeout into getCategoriesList is better. 
and no need for newMethod , i am sorry but its extra effort for nothing !

Reply:
and newMethod name is not descriptive, i cant guess what it should do by reading it . right ?

Reply:
it gives me an error on getSalesList(), fixed by using newMethod() this way

Reply:
fixed